palette/src/rgb/rgb.rs

```diff
@@ -953,14 +955,82 @@ where
     }
 }
 
+#[derive(Debug)]
+pub enum FromHexError {
+    ParseIntError(ParseIntError),
+    HexFormatError(&'static str)
+}
+
+impl From<ParseIntError> for FromHexError {
+    fn from(err: ParseIntError) -> FromHexError {
+        FromHexError::ParseIntError(err)
+    }
+}
+
+impl From<&'static str> for FromHexError {
+    fn from(err: &'static str) -> FromHexError {
+        FromHexError::HexFormatError(err)
+    }
+}
+
+impl<S: RgbStandard> FromStr for Rgb<S, u8> {
+    type Err = FromHexError;
+
+    // Parses a color hex code of format '#ff00bb' or '#abc' into a
+    // Rgb<S, u8> instance.
+    fn from_str(hex_code: &str) -> Result<Self, Self::Err> {
+        match hex_code.len() {
+            3 | 4 | 6 | 7 => {
+                let (red, green, blue, factor) = if hex_code.len() == 3 {
```

Comment:
I'm just stopping by to see how it's going. Looks promising! I would just recommend to either drop this if-else chain and only use the match, or the other way around. Both would work. As it is now, you are checking the length twice, at least. :slightly_smiling_face:

Reply:
An additional idea, to halve the amount of cases, is to check for the existence of `#` at the beginning first. If it's there, you slice it away. That way, you are down to only two possible lengths when parsing the components.

Reply:
Yes, I've done it this way. (December is a really busy month, :)

Reply:
Indeed, it is. But don't feel like you have to rush it or anything. :)